external-crates/move/documentation/book/src/enums.md

```diff
@@ -0,0 +1,335 @@
+# Enumerations
+
+An _enum_ is a user-defined data structure containing one-or-more _variants_. Each variant can
+optionally contain typed fields. The number, and types of these fields can differ for each variant
+in the enumeration. Fields in enums can store any non-reference, non-tuple type, including other
+structs or enums.
+
+As a simple example, consider the following enum definition in Move:
+
+```move
+public enum Action {
+    Stop,
+    Pause { duration: u32 },
+    MoveTo { x: u64, y: u64 },
+    Jump(u64),
+}
+```
+
+This declares an enum `Action` that represents different actions that can be taken by a game -- you
+can `Stop`, `Pause` for a given duration, `MoveTo` a specific location, or `Jump` to a specific
+height.
+
+Similarly to structs, enums can have [abilities](./abilities.md) that control what operations can be
+performed on them. It is important to note however that enums cannot have the `key` ability since
+they cannot be top-level objects.
+
+## Defining Enums
+
+Enums must be defined in a module, an enum must contain at least on variant, and each variant of an
+enum can either have no fields, positional fields, or named fields. Here are some examples of each:
+
+```move
+module a::m {
+    public enum Foo has drop {
+        VariantWithNoFields,
+        //                 ^ note: it is fine to have a trailing comma after variant declarations
+    }
+    public enum Bar has copy, drop {
+        VariantWithPositionalFields(u64, bool),
+    }
+    public enum Baz has drop {
+        VariantWithNamedFields { x: u64, y: bool, z: Bar },
+    }
+}
+```
+
+Enums cannot be recursive in any of their variants, so the following definitions of an enum are not
+allowed since they would be recursive in at least one variant:
```

Comment:
```suggestion
allowed because they would be recursive in at least one variant.

Incorrect:
```
Just a suggestion to maybe make more clear the incorrectness of these for those that scan instead of read.